homeassistant/components/switcher_kis/config_flow.py

```diff
@@ -0,0 +1,49 @@
+"""Config flow for Switcher integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DATA_DISCOVERY, DOMAIN
+from .utils import async_discover_devices
+
+
+class SwitcherFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
```

Comment:
You should be able to register a discovery config flow https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/config_entry_flow.py#L104

Reply:
I was not aware of this, I checked it and it can work but the current implementation will have better user experience since the discovery runs in background while waiting to the user to approve the setup. I suggest to keep it like this for now, I think there might also be an option to change core discovery flow to run in background. I will discuss with you offline and if needed make a separate PR